go.mod

```diff
@@ -10,4 +10,10 @@ require (
 	github.com/pkg/browser v0.0.0-20210911075715-681adbf594b8
 )
 
-require golang.org/x/sys v0.5.0 // indirect
+require (
+	github.com/davecgh/go-spew v1.1.1 // indirect
+	github.com/pmezard/go-difflib v1.0.0 // indirect
+	github.com/stretchr/testify v1.9.0 // indirect
+	golang.org/x/sys v0.5.0 // indirect
+	gopkg.in/yaml.v3 v3.0.1 // indirect
```

Comment:
I am very hesitant to accept new dependencies for any MSAL. There has to be a very strong reason to add a new dependency. 

I am not sure GoLang has a concept of "test dependencies" simialar to Java and Node or a clean separation of product vs test projects like .NET. So any test dependency today is seen as a product dependency. Unless the sitation has evolved @chlowell ?

As such, I consider all of the above as product depedencies, which provide an unacceptable burden to MSAL GO. 

- dependencies can have security issues. At worst, it creates a security vulnerability in MSAL Go, and possible in services using them. At "best", it's a just a matter of upgrading that dependency to a version that's not affected. Either way, it takes a LOT of time. 
- All dependencies need to be registered with https://repos.opensource.microsoft.com/ and by the looks of some of them, it's not going to happen. Even registering 4 new dependencies is going to take a long time.
![image](https://github.com/user-attachments/assets/2ea01000-3372-42f5-9bf2-3a51e36a9f38)

Reply:
> I am not sure GoLang has a concept of "test dependencies" simialar to Java and Node or a clean separation of product vs test projects like .NET. So any test dependency today is seen as a product dependency. Unless the sitation has evolved @chlowell ?

Go does separate the dependencies required to build a module from those required to run its tests. When an application installs a module, it also installs that module's build dependencies, but not its test dependencies. So, packages imported only in MSAL Go's `*_test.go` files aren't distributed to users. Unfortunately, as you can see, this distinction isn't visible in `go.mod`.